### PR TITLE
Create Cosmos DB account with advanced threat protection template

### PR DESCRIPTION
**Pull request by neryaco**
### Best Practice Checklist
Check these items before submitting a PR... See the Contribution Guide for the full detail: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/README.md 

1. uri's compatible with all clouds (Stack, China, Government)
1. Staged artifacts use _artifactsLocation & _artifactsLocationSasToken
1. Use a parameter for resource locations with the defaultValue set to resourceGroup().location
1. Folder names for artifacts (nestedtemplates, scripts, DSC)
1. Use literal values for apiVersion (no variables)
1. Parameter files (GEN-UNIQUE for value generation and no "changemeplease" values)
1. $schema and other uris use https
1. Use uniqueString() whenever possible to generate names for resources.  While this is not required, it's one of the most common failure points in a deployment. 
1. Update the metadata.json with the current date

For details: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md

- [x] - Please check this box once you've submitted the PR if you've read through the Contribution Guide and best practices checklist.

### Changelog

*
*
*

### CODE REVIEW

File: 201-cosmosdb-advanced-threat-protection-create-account/azuredeploy.json [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+          ],
+          "properties": {
+            "isEnabled": true
+          }
+        }
+      ]
+    }
+  ],
+  "outputs": {
+    "storageAccountName": {
```
Comment (tomerlmsft) [inline 201-cosmosdb-advanced-threat-protection-create-account/azuredeploy.json]: storageAccountName?

File: 201-cosmosdb-advanced-threat-protection-create-account/README.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
+# Create an Azure CosmosDB Account with Advanced Threat Protection enabled
```
Comment (tomerlmsft) [inline 201-cosmosdb-advanced-threat-protection-create-account/README.md]: we must add the preview disclaimer, e.g. (preview)

File: 201-cosmosdb-advanced-threat-protection-create-account/azuredeploy.json [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+      "metadata": {
+        "description": "Enable or disable Advanced Threat Protection."
+      }
+    }
+  },
+  "resources": [
+    {
+      "name": "[parameters('name')]",
+      "type": "Microsoft.DocumentDB/databaseAccounts",
+      "apiVersion": "2016-03-31",
```
Comment (tomerlmsft) [inline 201-cosmosdb-advanced-threat-protection-create-account/azuredeploy.json]: are you sure about the API version?

File: 201-cosmosdb-advanced-threat-protection-create-account/metadata.json [hunk 4]
```diff
@@ -0,0 +1,9 @@
+{
+    "$schema": "https://aka.ms/azure-quickstart-templates-metadata-schema#",
+    "type": "QuickStart",
+    "itemDisplayName": "Create a CosmosDB Account with Advanced Threat Protection.",
```
Comment (tomerlmsft) [inline 201-cosmosdb-advanced-threat-protection-create-account/metadata.json]: (preview)

File: 201-cosmosdb-advanced-threat-protection-create-account/azuredeploy.json [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+      "location": "[parameters('location')]",
+      "properties": {
+        "locations": [
+          {
+            "locationName": "[parameters('location')]"
+          }
+        ],
+        "databaseAccountOfferType": "[parameters('tier')]"
```
Comment (tomerlmsft) [inline 201-cosmosdb-advanced-threat-protection-create-account/azuredeploy.json]: this field called 'tier' at the top of the file, why there is a difference?

File: 201-cosmosdb-advanced-threat-protection-create-account/README.md [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+
```
Comment (tomerlmsft) [inline 201-cosmosdb-advanced-threat-protection-create-account/README.md]: what about links to the documentation?
is it common in arm templates?

File: 201-cosmosdb-advanced-threat-protection-create-account/azuredeploy.json (hunk 3 above)
Comment (neryaco) [reply 201-cosmosdb-advanced-threat-protection-create-account/azuredeploy.json]: same as all other cosmos db templates

File: 201-cosmosdb-advanced-threat-protection-create-account/metadata.json (hunk 4 above)
Comment (neryaco) [reply 201-cosmosdb-advanced-threat-protection-create-account/metadata.json]: itemDisplayName is limited to 60 chars, and the current text is 58 chars.

File: 201-cosmosdb-advanced-threat-protection-create-account/azuredeploy.json (hunk 5 above)
Comment (neryaco) [reply 201-cosmosdb-advanced-threat-protection-create-account/azuredeploy.json]: mostly to be consistent with the basic cosmos template. at the top is a parameter from user (we can name it as we want) and this one is the field name in the json object to be deployed.

File: 201-cosmosdb-advanced-threat-protection-create-account/README.md (hunk 6 above)
Comment (neryaco) [reply 201-cosmosdb-advanced-threat-protection-create-account/README.md]: in our template for storage we have a link. do we we have such link for cosmos to be used here?

File: 201-cosmosdb-advanced-threat-protection-create-account/azuredeploy.json (hunk 3 above)
Comment (neryaco) [reply 201-cosmosdb-advanced-threat-protection-create-account/azuredeploy.json]: @bmoore-msft this is the latest API version, but the CI pipeline fails with
`apiVersions-Should-Be-Recent.test.ps1 : Api versions should be under 2 years old (730 days) - (Microsoft.DocumentDB/databaseAccounts is 1202 days old)` 
can we use this version although it's old? (as it's the most recent)

Comment (bmoore-msft) [reply 201-cosmosdb-advanced-threat-protection-create-account/azuredeploy.json]: yes

**Pull request merged**